src/components/MessageInput.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useRef } from 'react';
+import { IMessage_Date, ITodo } from '../interfaces';
+
+type MessageInputProps = {
+  todos: ITodo[],
+  addMessage(msg: IMessage_Date): void
+}
+
+export const MessageInput: React.FC<MessageInputProps> = ({
+  todos,
+  addMessage,
+}) => {
+
+  const setNewMessage = (input_text:string): void => {
+    const rightNow = new Date();
+    const time = rightNow.toString().split('').splice(16, 5);
+    const timeString = time[0] + time[1] + time[2] + time[3] + time[4];
+    const new_message_date: IMessage_Date = {
+      message: input_text,
+      date: timeString,
+      id: Date.now(),
+    };
+
+    ref.current!.value = '';
```

Comment:
instead of using ref.current.value for input value, isn't it better to use state, setState
https://www.pluralsight.com/guides/how-to-use-react-to-set-the-value-of-an-input
https://www.google.com/search?q=input+value+react+state&oq=input+value+react+stat&aqs=chrome.0.0i19j69i57j0i19i22i30l5.7015j0j7&sourceid=chrome&ie=UTF-8